File: code/c9196.pas
program Bai196;
  var n, s, i: integer;
  begin
  readln(n);
  s := 0;
  for i := 1 to n div 2 do 
     if n mod i = 0 then  s:= s +i;
     if s = n then writeln('YES') else write('NO');
end..

File: code/c9196.cpp
#include <iostream>
using namespace std;

int main() 
{
    int n,x,s;
    cin>>n;
    s=0;
    for (x=1; x<=n/2; x++){
        if (n%x==0)
            s=s+x;
    }
    if (s==n)
      cout<<"YES";
    else cout<<"NO";
    return 0;
}